src/libraries/Microsoft.PowerFx.Core/Parser/TexlParser.cs

```diff
@@ -317,22 +322,30 @@ private ParseUserDefinitionResult ParseUDFsAndNamedFormulas(string script, Parse
                         }
 
                         // Parse expression
+                        definitionBeforeTrivia.Add(ParseTrivia());
                         var result = ParseExpr(Precedence.None);
+
                         if (result is TypeLiteralNode typeLiteralNode)
                         {
-                            if (typeLiteralNode.IsValid(out _))
+                            if (typeLiteralNode.IsValid(out var _))
                             {
                                 definedTypes.Add(new DefinedType(thisIdentifier.As<IdentToken>(), typeLiteralNode, true));
+                                userDefinitionSourceInfos.Add(new UserDefinitionSourceInfo(index++, UserDefinitionType.DefinedType, thisIdentifier.As<IdentToken>(), declaration, new SourceList(definitionBeforeTrivia), GetExtraTriviaSourceList()));
                             }
                             else
                             {
                                 definedTypes.Add(new DefinedType(thisIdentifier.As<IdentToken>(), typeLiteralNode, false));

```

Comment:
>false [](http://example.com/codeflow?start=115&length=5)

This is the only difference between this line and 332, it can be rewritten in a more concise way as `definedTypes.Add(new DefinedType(thisIdentifier.As<IdentToken>(), typeLiteralNode, typeLiteralNode.IsValid(out var _)))`